### PR TITLE
header values are be expected to be W3C baggage encoded

### PR DESCRIPTION
**Pull request by zeitlinger**
Blocked by https://github.com/open-telemetry/opentelemetry-specification/issues/3832

### CODE REVIEW

**Comment by zeitlinger** [comment]
@jack-berg can this error be ignored?

**Comment by jack-berg** [comment]
> @jack-berg can this error be ignored?

Yes and no. The codecov target is a goal, not a strict requirement. Sometimes its not practical to have all the code paths covered. Although in this case it seems like it should be straight forward to trigger than exception, no?

File: exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpConfigUtil.java [hunk 1]
```diff
@@ -89,7 +91,17 @@ public static void configureOtlpExporterBuilder(
...
-    headers.forEach(addHeader);
+    for (Map.Entry<String, String> entry : headers.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      try {
+        // headers are encoded as URL - see
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specifying-headers-via-environment-variables
+        addHeader.accept(key, URLDecoder.decode(value, "UTF-8"));
+      } catch (UnsupportedEncodingException e) {
+        throw new ConfigurationException("cannot decode header value: " + value, e);
```
Comment (jack-berg) [inline exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpConfigUtil.java:102]:
```suggestion
        throw new ConfigurationException("Cannot decode header value: " + value, e);
```

File: exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpConfigUtil.java [hunk 2]
```diff
@@ -89,7 +91,17 @@ public static void configureOtlpExporterBuilder(
...
       headers = config.getMap("otel.exporter.otlp.headers");
     }
-    headers.forEach(addHeader);
+    for (Map.Entry<String, String> entry : headers.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      try {
+        // headers are encoded as URL - see
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specifying-headers-via-environment-variables
+        addHeader.accept(key, URLDecoder.decode(value, "UTF-8"));
```
Comment (jack-berg) [inline exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/internal/OtlpConfigUtil.java:100]:
```suggestion
        addHeader.accept(key, URLDecoder.decode(value, StandardCharsets.UTF_8.displayName()));
```

**Review by jack-berg — COMMENTED** [review]
Thanks for the bug fix. Looks good, but let's try to cover the exception case.

**Comment by zeitlinger** [comment]
> Thanks for the bug fix. Looks good, but let's try to cover the exception case.

Fond a way :smile:

**Review by jack-berg — APPROVED** [review]

**Comment by jack-berg** [comment]
Waiting to merge until its clear whether this is going to be considered a breaking change or a bug fix based on this spec conversation: https://github.com/open-telemetry/opentelemetry-specification/issues/3832

**Pull request merged**